### PR TITLE
[

### PR DESCRIPTION
**Pull request by mklopp**

### CODE REVIEW

File: server/src/main/java/org/opentosca/toscana/plugins/cloudformation/util/CloudFormationScript.java [hunk 1]
```diff
@@ -10,11 +10,16 @@
 
 import static org.opentosca.toscana.plugins.cloudformation.CloudFormationModule.FILEPATH_TARGET;
 
+/**
+ Class that overrides the {@link BashScript} implementation.
```
Comment (hnicke) [inline server/src/main/java/org/opentosca/toscana/plugins/cloudformation/util/CloudFormationScript.java]: This comment does not provide anything, it just repeats the class header. A javadoc  comment should not explain `what` or `how` (if you're code fails to provide this information, refactor your code), it should always focus on the `why`.

File: server/src/main/java/org/opentosca/toscana/plugins/cloudformation/util/CloudFormationScript.java [hunk 2]
```diff
@@ -10,11 +10,16 @@
...
+ */
 public class CloudFormationScript extends BashScript {
 
     private final static Logger logger = LoggerFactory.getLogger(BashScript.class);
 
     /**
+     Standard constructor.
```
Comment (hnicke) [inline server/src/main/java/org/opentosca/toscana/plugins/cloudformation/util/CloudFormationScript.java]: I don't get how the  information `Standard constructor` provides any benefits. Could you explain?

File: server/src/main/java/org/opentosca/toscana/plugins/cloudformation/util/FileUpload.java [hunk 3]
```diff
@@ -3,27 +3,77 @@
 import java.util.ArrayList;
 import java.util.List;
 
+/**
+ Class that holds information about a file that needs to be uploaded.
+ */
 public class FileUpload {
+    /**
+     The file path where the file is stored.
```
Comment (hnicke) [inline server/src/main/java/org/opentosca/toscana/plugins/cloudformation/util/FileUpload.java]: Captain obvious? :D

File: server/src/main/java/org/opentosca/toscana/plugins/cloudformation/util/FileUpload.java [hunk 4]
```diff
@@ -3,27 +3,77 @@
...
+     */
     private final UploadFileType fileType;
 
+    /**
+     The different types a {@link FileUpload} can have.
+     */
     public enum UploadFileType {
-        FROM_CSAR, UTIL, OTHER
+        /**
+         All files can come from the CSAR and need to be uploaded have this type.
```
Comment (hnicke) [inline server/src/main/java/org/opentosca/toscana/plugins/cloudformation/util/FileUpload.java]: Could you rephrase this?

File: server/src/main/java/org/opentosca/toscana/plugins/cloudformation/util/MappingUtils.java [hunk 5]
```diff
@@ -8,21 +8,58 @@
...
+     */
     public static List<Integer> getCpuByMem(Integer memSize, ImmutableList<CapabilityMapper.InstanceType> instanceTypes) {
         return instanceTypes.stream()
             .filter(u -> u.getMemSize().equals(memSize))
             .map(CapabilityMapper.InstanceType::getNumCpus)
             .collect(Collectors.toList());
     }
 
+    /**
+     Gets the {@link String} representation of an {@link org.opentosca.toscana.plugins.cloudformation.mapper.CapabilityMapper.InstanceType}
```
Comment (hnicke) [inline server/src/main/java/org/opentosca/toscana/plugins/cloudformation/util/MappingUtils.java]: Don't overuse links. While they are useful, they make reading the source code more difficult. Linking to `String` does not make any sense. Every developer knows what you mean by that. At most, use `{@code String}`.

Btw you can also import the `InstanceType`.  This would also increase the readability.

File: server/src/main/java/org/opentosca/toscana/plugins/cloudformation/util/MappingUtils.java [hunk 6]
```diff
@@ -8,21 +8,58 @@
...
+     that has both {@code numCpus} and {@code memSize} as values.
+
+     @param numCpus       the number of cpus the {@link org.opentosca.toscana.plugins.cloudformation.mapper.CapabilityMapper.InstanceType}
```
Comment (hnicke) [inline server/src/main/java/org/opentosca/toscana/plugins/cloudformation/util/MappingUtils.java]: Yes, definitely import `InstanceType`.

File: server/src/main/java/org/opentosca/toscana/plugins/cloudformation/util/MappingUtils.java [hunk 7]
```diff
@@ -35,13 +72,13 @@ public static String getInstanceType(Integer numCpus, Integer memSize, Immutable
     }
 
     /**
-     Check if the value is in the list checker, if not take the next bigger. If there is none throw an
-     IllegalArgumentException
+     Checks if the value is in the {@link List} {@code checker}, if not take the next bigger value. If there is none
```
Comment (hnicke) [inline server/src/main/java/org/opentosca/toscana/plugins/cloudformation/util/MappingUtils.java]: Don't link to java's `List` - that's just clatter.

File: server/src/main/java/org/opentosca/toscana/plugins/cloudformation/mapper/CapabilityMapper.java [hunk 8]
```diff
@@ -303,25 +355,67 @@ private String findCombination(Integer numCpus, Integer memSize, ImmutableList<I
         return instanceType;
     }
 
+    /**
+     Class that represents a instance type of a EC2 or RDS.
```
Comment (mmunozba) [inline server/src/main/java/org/opentosca/toscana/plugins/cloudformation/mapper/CapabilityMapper.java]: `a instance type` -> `an instance type`

File: server/src/main/java/org/opentosca/toscana/plugins/cloudformation/visitor/CheckModelNodeVisitor.java [hunk 9]
```diff
@@ -18,24 +18,36 @@
...
+    /**
+     {@link Apache} node is supported.
+     */
     @Override
     public void visit(Apache node) {
         // noop
     }
 
+    /**
+     {@link Compute} node is supported.
```
Comment (mmunozba) [inline server/src/main/java/org/opentosca/toscana/plugins/cloudformation/visitor/CheckModelNodeVisitor.java]: Not sure if the "X node is supported"-comments are really needed here. 
The `Only Linux, Ubuntu is supported!` part is the important one.

File: server/src/main/java/org/opentosca/toscana/plugins/cloudformation/visitor/CheckModelNodeVisitor.java [hunk 10]
```diff
@@ -18,24 +18,36 @@
...
+     <br>
+     Only linux, ubuntu is supported!
```
Comment (mmunozba) [inline server/src/main/java/org/opentosca/toscana/plugins/cloudformation/visitor/CheckModelNodeVisitor.java]: `linux, ubuntu` -> `Linux, Ubuntu`

File: server/src/main/java/org/opentosca/toscana/plugins/cloudformation/visitor/CheckModelRelationshipVisitor.java [hunk 11]
```diff
@@ -10,20 +10,29 @@
 import org.opentosca.toscana.model.visitor.UnsupportedTypeException;
 
 /**
- Class for checking the models relationships
+ Class for checking the models relationships whether they are supported or not.
```
Comment (mmunozba) [inline server/src/main/java/org/opentosca/toscana/plugins/cloudformation/visitor/CheckModelRelationshipVisitor.java]: `models` -> `model's`

File: server/src/main/java/org/opentosca/toscana/plugins/cloudformation/visitor/CheckModelNodeVisitor.java [hunk 12]
```diff
@@ -18,24 +18,36 @@
 import com.google.common.collect.Lists;
 
 /**
- Class for checking the models nodes
+ Class for checking the models nodes whether they are supported or not.
+ <br>
+ Methods that are not overridden will throw an {@link UnsupportedTypeException}.
```
Comment (mmunozba) [inline server/src/main/java/org/opentosca/toscana/plugins/cloudformation/visitor/CheckModelNodeVisitor.java]: You should probably explain why this is the case.

File: server/src/main/java/org/opentosca/toscana/plugins/cloudformation/visitor/CheckModelRelationshipVisitor.java [hunk 13]
```diff
@@ -10,20 +10,29 @@
...
+ <br>
+ Methods that are not overridden will throw an {@link UnsupportedTypeException}.
```
Comment (mmunozba) [inline server/src/main/java/org/opentosca/toscana/plugins/cloudformation/visitor/CheckModelRelationshipVisitor.java]: You should probably explain why this is the case.

File: server/src/main/java/org/opentosca/toscana/plugins/cloudformation/visitor/CloudFormationVisitor.java [hunk 14]
```diff
@@ -25,27 +25,57 @@
...
+ */
+public abstract class CloudFormationVisitor {
 
     protected final Logger logger;
+    /**
+     The topology of the {@link org.opentosca.toscana.model.EffectiveModel} used in the transformation.
+     */
     protected Graph<RootNode, RootRelationship> topology;
+    /**
+     The CloudFormationModule that holds every information during the transformation.
```
Comment (mmunozba) [inline server/src/main/java/org/opentosca/toscana/plugins/cloudformation/visitor/CloudFormationVisitor.java]: `every information` -> `all the information`

File: server/src/main/java/org/opentosca/toscana/plugins/cloudformation/visitor/PrepareModelNodeVisitor.java [hunk 15]
```diff
@@ -31,15 +31,28 @@
...
      */
     public PrepareModelNodeVisitor(TransformationContext context, CloudFormationModule cfnModule) {
         super(context, cfnModule);
     }
 
+    /**
+     Sets values to default values that are not required by the {@link org.opentosca.toscana.model.EffectiveModel} but by
+     CloudFormation.
+     <br>
+     Passwords need to have a length of 8. If this is not given there is a random password created.
```
Comment (mmunozba) [inline server/src/main/java/org/opentosca/toscana/plugins/cloudformation/visitor/PrepareModelNodeVisitor.java]: Maybe rephrase the second sentence to make clear that the random password is used instead of the normal password, like:
`If no password is given, a randomly created password is used instead.`
